### ISSUE TITLE
setting Socket's tcpNoDelay option returns null

### ISSUE THREAD

**Issue by isoos** [dart-lang/sdk#34983]
`Dart VM version: 2.1.0-dev.5.0 (Wed Sep 19 19:15:19 2018 +0200) on "macos_x64"`

```
final socket = await Socket.connect(host, port);
final result = socket.setOption(SocketOption.tcpNoDelay, true);
print(result);
```

According to the [dartdoc](https://api.dartlang.org/stable/2.0.0/dart-io/Socket/setOption.html), this method should return a non-null bool, but it returned a null.

**Comment by willyabs** [comment]
Use SecureSocket also having the same issue.

Additionally, when client SecureSocket continually send messages util 4096th byte, the message is separated into two parts to server. For example, an input message is 100 bytes, when client have sent 41 messages, the 41th message that server received will be separated into 96 bytes and 4 bytes. Maybe this issue also related to SocketOption.tcpNoDelay.

**Comment by lrhn** [comment]
The first issue seems like a simple mistake in `socket_patch.dart`. I'll make a fix.

The second issue should probably be tracked as a separate bug.

**Comment by sortie** [comment]
Fixed by Lasse in e79cb0232f67d8a1c172b1833570166430f2cd37.
